tests/tl/test_differential_expression_analysis.py

```diff
@@ -0,0 +1,412 @@
+        pd.DataFrame(
+            [[True], [False]],
+            columns=[DeaColumns.SIGNIFICANT],
+            index=["gene1", "gene2"],
+        )
+    )
+
+
+def test_dea_get_significance_static():
+    result = pd.DataFrame(
+        [[0.055, 2, 0.009], [0.1, 3, 0.1]],
+        columns=[DeaColumns.PVALUE, DeaColumns.LOG2FC, DeaColumns.QVALUE],
+        index=["gene1", "gene2"],
+    )
+    significant = DifferentialExpressionAnalysis.get_significance(result, 0.01)
+    assert significant.equals(
+        pd.DataFrame(
+            [[True], [False]],
+            columns=[DeaColumns.SIGNIFICANT],
+            index=["gene1", "gene2"],
+        )
+    )
+
+
+def test_dea_get_dict_key_from_instance():
+    input_data = pd.DataFrame()
+    dea = TestableDifferentialExpressionAnalysis(input_data)
+    assert (
+        dea.get_dict_key({DeaParameters.METADATA: pd.DataFrame()})
+        == """{'metadata': Empty DataFrame
+Columns: []
+Index: []}"""
+    )
+
+
+def test_dea_get_dict_key_static():
+    assert (
+        DifferentialExpressionAnalysis.get_dict_key(
+            {DeaParameters.METADATA: pd.DataFrame()}
+        )
+        == """{'metadata': Empty DataFrame
+Columns: []
+Index: []}"""
+    )
+
+
+class TestableDifferentialExpressionAnalysisTwoGroups(
+    DifferentialExpressionAnalysisTwoGroups
+):
+    @staticmethod
+    def _allowed_parameters():
+        return [
+            DeaParameters.METADATA,
+            DeaParameters.GROUP1,
+            DeaParameters.GROUP2,
+            DeaParameters.GROUPING_COLUMN,
+        ]
+
+    def _run_statistical_test(self, **kwargs):
+        group1, group2 = self._get_group_members(kwargs)
+        return self._statistical_test_fun(self.input_data, group1=group1, group2=group2)
+
+    @staticmethod
+    def _statistical_test_fun(input_data, group1, group2):
+        return pd.DataFrame(
+            [[0.055, 2, 0.009]],
+            columns=[DeaColumns.PVALUE, DeaColumns.LOG2FC, DeaColumns.QVALUE],
+            index=["gene1"],
+        )
+
+    valid_parameter_input = {
+        DeaParameters.METADATA: pd.DataFrame(
+            [
+                ["sample1", "group1"],
+                ["sample2", "group1"],
+                ["sample3", "group2"],
+                ["sample4", "group2"],
+            ],
+            columns=[Cols.SAMPLE, "grouping_column"],
+        ),
+        DeaParameters.GROUP1: "group1",
+        DeaParameters.GROUP2: "group2",
+        DeaParameters.GROUPING_COLUMN: "grouping_column",
+    }
+
+    valid_data_input = pd.DataFrame(
+        [[1, 2, 3, 4], [2, 3, 4, 5], [3, 4, 5, 6]],
+        columns=["sample1", "sample2", "sample3", "sample4"],
+        index=["gene1", "gene2", "gene3"],
+    ).T
+
+
+def test_dea_two_groups_no_abstractmethods():
+    with pytest.raises(TypeError):
+        DifferentialExpressionAnalysisTwoGroups(None)
+
+
+def test_dea_two_groups_perform_success():
+    input_data = TestableDifferentialExpressionAnalysisTwoGroups.valid_data_input
```

Comment:
could we make `valid_data_input` a global constant and use it here and for defining `TestableDifferentialExpressionAnalysisTwoGroups`? Seeing `TestableDifferentialExpressionAnalysisTwoGroups` and an "real" instance at the same place is a bit misleadding.